### ISSUE TITLE
ffi: AuthenticationService treats sliding sync proxy as mandatory

### ISSUE THREAD

**Issue by Johennes** [matrix-org/matrix-rust-sdk#3579]
Calling [`configure_homeserver`](https://github.com/matrix-org/matrix-rust-sdk/blob/main/bindings/matrix-sdk-ffi/src/authentication_service.rs#L320) on an `AuthenticationService` instance currently results in an error if there is neither a custom sliding sync proxy nor a sliding sync proxy in the home server's .well-known. It would be great if login was possible on home servers that don't currently support sliding sync. Maybe this could be achieved through an additional parameter in the constructor of `AuthenticationService`?

**Comment by bnjbvr** [comment]
You're in luck! Sounds like something @pixlwave's been working on in https://github.com/matrix-org/matrix-rust-sdk/pull/3569 :eyes:

**Comment by poljar** [comment]
Would that help that much? Large parts of the bindings assume SS support, do we even expose the sync v2 methods?

**Comment by Johennes** [comment]
> [...] I'm now working on moving all the methods from `AuthenticationService` directly onto `Client`

Oh, should I be rebasing the changes from https://github.com/matrix-org/matrix-rust-sdk/pull/3558, too, then?

> Would that help that much? Large parts of the bindings assume SS support, do we even expose the sync v2 methods?

Good point. I haven't actually looked further into the bindings yet because I'm blocked on logging in due to the PR mentioned above. I suppose the v2 methods could be exposed as well if they're currently missing? Not sure how much hassle that would be though.

**Comment by poljar** [comment]
> Good point. I haven't actually looked further into the bindings yet because I'm blocked on logging in due to the PR mentioned above. I suppose the v2 methods could be exposed as well if they're currently missing? Not sure how much hassle that would be though.

Yeah, it's a good first step but I think you'll have to take a bunch of other steps sadly. Perhaps having the sync method is even enough to have something to play with.

Goodspeed on your quest.

**Comment by pixlwave** [comment]
> Oh, should I be rebasing the changes from #3558, too, then?

Ah yeah, I’ve just seen the PR, yes please (and sorry). I should have a PR up at some stage tomorrow that deletes the `AuthenticationService` entirely - it’s all in the name of testing all the logic that accumulated in this part of the FFI.
